runtime/src/global_request_scheduler_minheap.c

```diff
@@ -19,7 +19,8 @@ global_request_scheduler_minheap_add(void *sandbox_request)
 {
 	assert(sandbox_request);
 	assert(global_request_scheduler_minheap);
-	if (unlikely(!listener_thread_is_running())) panic("%s is only callable by the listener thread\n", __func__);
+	/* panic is not applicable here because all worker_threads can add a request to the global queue if it's a chain function */
+	//if (unlikely(!listener_thread_is_running())) panic("%s is only callable by the listener thread\n", __func__);
```

Comment:
I think that the check is still correct (how would a worker be running without the listener?), but the printout is very wrong. Something more appropriate would be `"%s is only callable after the listener thread has begun execution"`. I don't know if it adds much value as a check, though.

Reply:
This `global_request_scheduler_minheap_add` is adding a request to the global queue. In the original sledge, only the listener thread creates the request and adds it to the global queue, so `listener_thread_is_running()` won't panic. Right now, it's not only the listener thread can add a request to the queue, but all the worker threads can do that, so here `listener_thread_is_running()` will panic because the current thread might not be the listener thread.

`listener_thread_is_running()` checks if the current thread is the listener thread or not. Sledge will panic here if not comment on this line.

Reply:
I understand all of this. I think that I don't understand what `listener_thread_is_running` does. Given your response, it really tells us if the listener thread is currently the thread that is executing this thread, I assume. In that case, I understand your response. I thought it simply returned if the thread had been created or not. Thanks for the clarification.

Reply:
The name of the this function is indeed confusing. Should be something like: `is_this_listener_thread()`

Reply:
The original intent here was to panic if a worker called this function because I had perf concerns about multiple threads adding to the global structure.